core/flyout_base.js

```diff
@@ -300,6 +302,15 @@ Blockly.Flyout.prototype.init = function(targetWorkspace) {
   this.workspace_.setVariableMap(this.targetWorkspace.getVariableMap());
 
   this.workspace_.createPotentialVariableMap();
+
+  this.workspace_.getComponentManager().addComponent({
```

Comment:
Good catch. That was a mistake that I hadn't caught.

Fixing this revealed an unexpected problem with Flyouts.
There is an issue with how Flyouts compute `getClientRect` which cause it to only work for Flyouts with `autoClose=false`. I can elaborate further offline as its a bit complicated to explain.
I have 2 approaches that I think could work to avoid this issue for now, if you look at this commit you can see the 2 approaches: https://github.com/google/blockly/pull/4852/commits/b1b4886df1adb3b064b02ff1d9898e4e362046ba

Reply:
I'm not sure what the second approach is, but I think it makes sense to return if `autoClose` is true. Readability wise I think checking whether the `flyout.isVisible` would make more sense, but I'm assuming it wouldn't work because of how we record delete areas?